whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapRedirectTestIntegration.java

```diff
@@ -185,6 +190,96 @@ public void inetnum_outside_range() {
+                "source:       TEST");
+        databaseHelper.addObject("" +
+                "inetnum:      217.180.128.0 - 217.180.191.255\n" +
+                "netname:      TEST-NET-NAME\n" +
+                "source:       TEST");
+        ipTreeUpdater.rebuild();
+
+        addResourceData("test", "217.0.0.0 - 217.255.255.255");
+
+        refreshResourceData();
+        //217.180.192.0 - 217.180.255.255 is not allocated
+
+        final Ip ip = RestTest.target(getPort(), String.format("rdap/%s", "ip/217.180.0.0/16"))
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(Ip.class);
+
+        assertThat(ip.getHandle(), equalTo("217.0.0.0 - 217.255.255.255"));
+
+    }
+
+    @Test
+    public void inetnum_redirect_partial_delegation() {
+        databaseHelper.addObject("" +
+                "inetnum:      217.180.128.0 - 217.180.191.255\n" +
+                "netname:      TEST-NET-NAME\n" +
+                "source:       TEST");
+        ipTreeUpdater.rebuild();
+
+        addResourceData("one", "217.180.0.0 - 217.180.255.255");
+        addResourceData("test", "217.180.128.0 - 217.180.191.255");
+
+        refreshResourceData();
+        //217.180.192.0 - 217.180.255.255 is not allocated
+        try {
+            RestTest.target(getPort(), String.format("rdap/%s", "ip/217.180.0.0/16"))
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .get(String.class);
+            fail();
+        } catch (final RedirectionException e) {
+            assertThat(e.getResponse().getHeaders().getFirst("Location").toString(), is("https://rdap.one" +
+                    ".net/ip/217.180.0.0/16"));
+        }
+    }
+
+    @Test
+    public void inetnum_larger_assigment_redirect_partial_delegation() {
+        databaseHelper.addObject("" +
+                "inetnum:      217.180.128.0 - 217.180.191.255\n" +
+                "netname:      TEST-NET-NAME\n" +
+                "source:       TEST");
+        ipTreeUpdater.rebuild();
+
+        addResourceData("one", "217.0.0.0 - 217.255.255.255");
+        addResourceData("test", "217.180.128.0 - 217.180.191.255");
+
+        refreshResourceData();
+        //217.180.192.0 - 217.180.255.255 is not allocated
+        try {
+            RestTest.target(getPort(), String.format("rdap/%s", "ip/217.180.0.0/16"))
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .get(String.class);
+            fail();
+        } catch (final RedirectionException e) {
+            assertThat(e.getResponse().getHeaders().getFirst("Location").toString(), is("https://rdap.one" +
+                    ".net/ip/217.180.0.0/16"));
+        }
+    }
+    
+    @Test
+    public void inetnum_inside_range_not_found() {
+        try {
+            RestTest.target(getPort(), String.format("rdap/%s", "ip/192.0.0.1"))
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(String.class);
```

Comment:
Thanks for fixing the test! Can you please add a `fail()` here also in case no exception is thrown

Reply:
Done